### ISSUE TITLE
I need to extend the Session Expiry Timeout/ Login Expiry Timeout in ASP.NET Core Blazor Server

### ISSUE THREAD

**Issue by tahaAttique** [dotnet/aspnetcore#31232]
I am developing a Blazor Server ASP.net core app. It has Cookie based authentication using RevalidatingIdentityAuthenticationStateProvider. When I login I find '.AspNetCore.Identity.Application' Cookie in my browser.

The problem I am facing is that I want to expend Session Expiry Timeout/Login Expiry Timeout and I am unable to do so.

**Right Now Situation:** After 30 min of my login when I refresh the page it redirects to the login screen which shows that the session has been expired.
**What I want:** I want that either the session does not expire or expires after a really long time.

Help will be appreciated !

**Comment by javiercn** [comment]
@tahaAttique thanks for contacting us.

You need to trigger a fetch request from the browser to an endpoint on the server to make the cookie slide

**Comment by tahaAttique** [comment]
Adding the following lines in Startup.cs solved the problem
```
services.ConfigureApplicationCookie(ops =>
{
    ops.ExpireTimeSpan = TimeSpan.FromMinutes(30);
    ops.SlidingExpiration = true;
});
```

**Comment by tahaAttique** [comment]
Thanks @javiercn  !!

**Comment by tahaAttique** [comment]
![image](https://user-images.githubusercontent.com/47359465/112715927-2bf4bd00-8f05-11eb-9d2b-fc89e1a66ccb.png)

Tried this and found that we are able to lessen the expiry time but in case we make it greater than 30 min, It does not work. It still expires at 30 min.

**Comment by tahaAttique** [comment]
Hey @javiercn  ! Thanks for the help.
Is there no solution other than refreshing the cookie manually?

**Comment by SteveSandersonMS** [comment]
@tahaAttique No, there isn't another solution as far as I know. Cookies can only be issued when there's an HTTP response, so you need to make an HTTP request to get one.

We don't have a built-in thing in Blazor that sends "keepalive" HTTP requests because we don't know what sort of rules you'd want around frequency or whether to keep doing it forever even if the user seems to have become idle. If this ends up becoming a common customer request, we could look into designing some kind of built-in feature.
